sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -103,7 +103,7 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
         toSQLType(from),
         toSQLType(to),
         toSQLId(columnName)),
-      context = None,
+      context = null,
```

Comment:
I think we should never pass null as the context array to exception classes. If context is not available, we should pass `Array.empty`

Reply:
ditto to other places in this file.

Reply:
done